packages/quantic/force-app/main/default/lwc/quanticPager/e2e/quantic-pager.e2e.ts

```diff
@@ -0,0 +1,229 @@
+      });
+    });
+
+    test.describe('when on the last page', () => {
+      test.beforeEach(async ({pager, search}) => {
+        await pager.goToLastPage();
+        await search.waitForSearchResponse();
+      });
+
+      test('should disable the next page button', async ({pager}) => {
+        await expect(pager.pageButtons.nth(-1)).toHaveAttribute(
+          'aria-pressed',
+          'true'
+        );
+        await expect(pager.nextPageButton).toBeDisabled();
+      });
+    });
+
+    test.describe('when clicking the next page button', () => {
+      test('should trigger a new search and log analytics', async ({
+        pager,
+        search,
+      }) => {
+        await pager.clickNextPageButton();
+        const response = await search.waitForSearchResponse();
+        const {firstResult} = await response.request().postDataJSON();
+        await expect(firstResult).toBe(numberOfResultsPerPage);
+        await pager.waitForPagerUaAnalytics('pagerNext');
+      });
+    });
+
+    test.describe('when clicking the previous page button', () => {
+      test('should trigger a new search and log analytics', async ({
+        pager,
+        search,
+      }) => {
+        await pager.clickNextPageButton();
+        await search.waitForSearchResponse();
+
+        await pager.clickPreviousPageButton();
+        const response = await search.waitForSearchResponse();
+        const {firstResult} = await response.request().postDataJSON();
+        await expect(firstResult).toBe(0);
+        await pager.waitForPagerUaAnalytics('pagerPrevious');
+      });
+    });
+
+    test.describe('when clicking a specific page button', () => {
+      test('should trigger a new search and log analytics', async ({
+        pager,
+        search,
+      }) => {
+        const examplePage = 3;
+        await pager.clickPageNumberButton(examplePage);
+        const response = await search.waitForSearchResponse();
+        const {firstResult} = await response.request().postDataJSON();
+        await expect(firstResult).toBe(
+          numberOfResultsPerPage * (examplePage - 1)
+        );
+        await pager.waitForPagerUaAnalytics('pagerNumber');
+      });
+    });
+
+    test.describe('when a new search is made', () => {
+      test('should select the first page', async ({pager, search}) => {
+        await pager.clickNextPageButton();
+        await search.waitForSearchResponse();
+
+        await expect(pager.previousPageButton).not.toBeDisabled();
+        await search.performSearch();
+        const searchResponse = await search.waitForSearchResponse();
+        const {firstResult} = await searchResponse.request().postDataJSON();
+
+        const expectedPage = 0;
+        await expect(pager.pageButtons.nth(expectedPage)).toHaveAttribute(
+          'aria-pressed',
+          'true'
+        );
+        expect(firstResult).toBe(0);
+      });
+    });
+
+    if (useCase.value === 'search') {
```

Comment:
I think this I would have mostly on a "full search page" test since it's not really specific to the pager.